src/DomainGuest.sol

```diff
@@ -0,0 +1,374 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+/// DomainGuest.sol -- xdomain guest dss manager
+
+// Copyright (C) 2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./TeleportGUID.sol";
+import {DomainHost} from "./DomainHost.sol";
+
+interface VatLike {
+    function hope(address usr) external;
+    function debt() external view returns (uint256);
+    function Line() external view returns (uint256);
+    function file(bytes32 what, uint256 data) external;
+    function dai(address usr) external view returns (uint256);
+    function sin(address usr) external view returns (uint256);
+    function heal(uint256 rad) external;
+    function swell(address u, int256 rad) external;
+}
+
+interface TokenLike {
+    function balanceOf(address usr) external view returns (uint256);
+    function transferFrom(address src, address dst, uint256 wad) external returns (bool);
+    function approve(address usr, uint wad) external returns (bool);
+    function mint(address to, uint256 value) external;
+}
+
+interface DaiJoinLike {
+    function vat() external view returns (VatLike);
+    function dai() external view returns (TokenLike);
+    function join(address usr, uint256 wad) external;
+    function exit(address usr, uint256 wad) external;
+}
+
+interface EndLike {
+    function cage() external;
+    function debt() external view returns (uint256);
+}
+
+interface RouterLike {
+    function settle(bytes32 sourceDomain, bytes32 targetDomain, uint256 amount) external;
+    function registerMint(TeleportGUID calldata teleport) external;
+}
+
+struct Settlement {
+    bytes32 sourceDomain;
+    bytes32 targetDomain;
+    uint256 amount;
+    bool    sent;
+}
+
+/// @title Support for xchain MCD, canonical DAI and Maker Teleport - remote instance
```

Comment:
Not sure if the "remote instance" makes sense. As actually for the place where we deploy, this is the local contract and I think that is actually the wording for the ids, where local is for the actual contract and remote for the other side (whatever is host or guest). So maybe this should just be mentioned as "guest instance" and we should put "host instance"  in the other file.